### PR TITLE
Typing for uid-safe [Types 2.0]

### PR DESCRIPTION
**Pull request by geoffreak**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

---

Typing for [`uid-safe`](https://github.com/crypto-utils/uid-safe)

### CODE REVIEW

File: uid-safe/uid-safe-test.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as uid from 'uid-safe';
```
Comment (andy-ms) [inline uid-safe/uid-safe-test.ts]: File name should be `uid-safe-tests.ts`

File: uid-safe/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for co-body
```
Comment (andy-ms) [inline uid-safe/index.d.ts]: Wrong project name

File: uid-safe/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// Project: https://github.com/crypto-utils/uid-safe
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace UID {
+
+    export interface Generate {
+
+        (byteLenth: number, callback: (err: any, str: string) => any): void;
```
Comment (andy-ms) [inline uid-safe/index.d.ts]: byteLength

File: uid-safe/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        (byteLenth: number): Promise<any>;
+        (byteLenth: number, callback?: (err: any, str: string) => any): Promise<any> | void;
```
Comment (andy-ms) [inline uid-safe/index.d.ts]: This overload isn't very useful. The other two should be sufficient.

**Review by ghost — CHANGES_REQUESTED** [review]

File: uid-safe/index.d.ts (hunk 2 above)
Comment (geoffreak) [inline uid-safe/index.d.ts]: Whoops! Going too fast here

File: uid-safe/index.d.ts (hunk 4 above)
Comment (geoffreak) [inline uid-safe/index.d.ts]: Ah, out of habit I wrote the definition. I'll nix this.

**Comment by geoffreak** [comment]
Alright @andy-ms, I fixed those obvious mistakes

**Comment by andy-ms** [comment]
Also, could you include the version `// Type definitions for uid-safe v2.1`?

**Comment by geoffreak** [comment]
Version has been added

**Review by ghost — APPROVED** [review]

**Pull request merged**